Unstable test `testTopSQLReporter.TestCollectInternal`

## Bug Report

Please answer these questions before submitting your issue. Thanks!
```

[2021-07-18T12:10:23.616Z] FAIL: reporter_test.go:400: testTopSQLReporter.TestCollectInternal

[2021-07-18T12:10:23.616Z] 

[2021-07-18T12:10:23.616Z] reporter_test.go:432:

[2021-07-18T12:10:23.616Z]     c.Assert(exist, IsTrue)

[2021-07-18T12:10:23.616Z] ... obtained bool = false

[2021-07-18T12:10:23.616Z] 

[2021-07-18T12:10:23.617Z] [2021/07/18 20:08:34.168 +08:00] [WARN] [reporter.go:272] ["[top-sql] data points is invalid, it should never happen"] [self="{\"SQLDigest\":null,\"PlanDigest\":null,\"TimestampList\":null,\"CPUTimeMsList\":null,\"CPUTimeMsTotal\":0}"] [evict="{\"SQLDigest\":null,\"PlanDigest\":null,\"TimestampList\":[1],\"CPUTimeMsList\":[10,30],\"CPUTimeMsTotal\":0}"]

[2021-07-18T12:10:23.617Z] OOPS: 6 passed, 1 FAILED

[2021-07-18T12:10:23.617Z] --- FAIL: TestT (4.69s)
```

### 1. Minimal reproduce step (Required)

in ci https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/18263/pipeline
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/server/tidb_test.go
+++ b/server/tidb_test.go
@@ -1758,9 +1758,9 @@ func (ts *tidbTestTopSQLSuite) TestTopSQLAgent(c *C) {
 		records := agentServer.GetLatestRecords()
 		c.Assert(len(records), Equals, n)
 		for _, r := range records {
-			sqlNormalized, exist := agentServer.GetSQLMetaByDigestBlocking(r.SqlDigest, time.Second)
+			sqlMeta, exist := agentServer.GetSQLMetaByDigestBlocking(r.SqlDigest, time.Second)
 			c.Assert(exist, IsTrue)
-			c.Check(sqlNormalized, Matches, "select.*from.*join.*")
+			c.Check(sqlMeta.NormalizedSql, Matches, "select.*from.*join.*")
 			if len(r.PlanDigest) == 0 {
 				continue
 			}

--- a/util/topsql/reporter/mock/server.go
+++ b/util/topsql/reporter/mock/server.go
@@ -140,21 +140,14 @@ func (svr *mockAgentServer) WaitCollectCnt(cnt int, timeout time.Duration) {
 	}
 }
 
-func (svr *mockAgentServer) GetSQLMetaByDigest(digest []byte) (tipb.SQLMeta, bool) {
-	svr.Lock()
-	sqlMeta, exist := svr.sqlMetas[string(digest)]
-	svr.Unlock()
-	return sqlMeta, exist
-}
-
-func (svr *mockAgentServer) GetSQLMetaByDigestBlocking(digest []byte, timeout time.Duration) (normalizedSQL string, exist bool) {
+func (svr *mockAgentServer) GetSQLMetaByDigestBlocking(digest []byte, timeout time.Duration) (meta tipb.SQLMeta, exist bool) {
 	start := time.Now()
 	for {
 		svr.Lock()
 		sqlMeta, exist := svr.sqlMetas[string(digest)]
 		svr.Unlock()
 		if exist || time.Since(start) > timeout {
-			return sqlMeta.NormalizedSql, exist
+			return sqlMeta, exist
 		}
 		time.Sleep(time.Millisecond)
 	}

--- a/util/topsql/reporter/reporter_test.go
+++ b/util/topsql/reporter/reporter_test.go
@@ -121,9 +121,9 @@ func (s *testTopSQLReporter) TestCollectAndSendBatch(c *C) {
 		for i := range req.RecordListTimestampSec {
 			c.Assert(req.RecordListTimestampSec[i], Equals, uint64(1))
 		}
-		normalizedSQL, exist := agentServer.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
+		sqlMeta, exist := agentServer.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
 		c.Assert(exist, IsTrue)
-		c.Assert(normalizedSQL, Equals, "sqlNormalized"+strconv.Itoa(id))
+		c.Assert(sqlMeta.NormalizedSql, Equals, "sqlNormalized"+strconv.Itoa(id))
 		normalizedPlan, exist := agentServer.GetPlanMetaByDigestBlocking(req.PlanDigest, time.Second)
 		c.Assert(exist, IsTrue)
 		c.Assert(normalizedPlan, Equals, "planNormalized"+strconv.Itoa(id))
@@ -165,9 +165,9 @@ func (s *testTopSQLReporter) TestCollectAndEvicted(c *C) {
 		}
 		c.Assert(id > maxSQLNum, IsTrue)
 		c.Assert(req.RecordListCpuTimeMs[0], Equals, uint32(id))
-		normalizedSQL, exist := agentServer.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
+		sqlMeta, exist := agentServer.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
 		c.Assert(exist, IsTrue)
-		c.Assert(normalizedSQL, Equals, "sqlNormalized"+strconv.Itoa(id))
+		c.Assert(sqlMeta.NormalizedSql, Equals, "sqlNormalized"+strconv.Itoa(id))
 		normalizedPlan, exist := agentServer.GetPlanMetaByDigestBlocking(req.PlanDigest, time.Second)
 		c.Assert(exist, IsTrue)
 		c.Assert(normalizedPlan, Equals, "planNormalized"+strconv.Itoa(id))
@@ -428,7 +428,7 @@ func (s *testTopSQLReporter) TestCollectInternal(c *C) {
 		if id == 0 {
 			c.Fatalf("the id should not be 0")
 		}
-		sqlMeta, exist := agentServer.GetSQLMetaByDigest(req.SqlDigest)
+		sqlMeta, exist := agentServer.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
 		c.Assert(exist, IsTrue)
 		c.Assert(sqlMeta.IsInternalSql, Equals, id%2 == 0)
 	}

--- a/util/topsql/topsql_test.go
+++ b/util/topsql/topsql_test.go
@@ -170,11 +170,11 @@ func (s *testSuite) TestTopSQLReporter(c *C) {
 		c.Assert(len(req.RecordListCpuTimeMs) > 0, IsTrue)
 		c.Assert(req.RecordListCpuTimeMs[0] > 0, IsTrue)
 		c.Assert(req.RecordListCpuTimeMs[0] > 0, IsTrue)
-		normalizedSQL, exist := server.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
+		sqlMeta, exist := server.GetSQLMetaByDigestBlocking(req.SqlDigest, time.Second)
 		c.Assert(exist, IsTrue)
 		expectedNormalizedSQL, exist := sqlMap[string(req.SqlDigest)]
 		c.Assert(exist, IsTrue)
-		c.Assert(normalizedSQL, Equals, expectedNormalizedSQL)
+		c.Assert(sqlMeta.NormalizedSql, Equals, expectedNormalizedSQL)
 
 		expectedNormalizedPlan := sql2plan[expectedNormalizedSQL]
 		if expectedNormalizedPlan == "" || len(req.PlanDigest) == 0 {

